#! /bin/sh -e
## 21_fix_mips_mipsel.dpatch by Martin Loschwitz <madkiss@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Make it compile on mips / mipsel

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@

diff -ruN qt-x11-free-3.3.3.old/src/qt.pro qt-x11-free-3.3.3/src/qt.pro
--- qt-x11-free-3.3.3.old/src/qt.pro	2004-05-14 13:55:30.000000000 +0000
+++ qt-x11-free-3.3.3/src/qt.pro	2004-11-14 17:34:27.000000000 +0000
@@ -9,6 +9,8 @@
 CONFIG		+= qt warn_on depend_includepath
 CONFIG          += qmake_cache
 
+QMAKE_CXXFLAGS += "-Wa,-xgot"
+
 win32:!shared:CONFIG += staticlib
 
 win32-borland {
diff -ruN qt-x11-free-3.3.3.old/src/qt.pro qt-x11-free-3.3.3/src/qt.pro
--- qt-x11-free-3.3.3.old/src/qt.pro	2004-05-14 13:55:30.000000000 +0000
+++ qt-x11-free-3.3.3/src/qt.pro	2004-11-16 18:35:19.000000000 +0000
@@ -9,6 +9,9 @@
 CONFIG		+= qt warn_on depend_includepath
 CONFIG          += qmake_cache
 
+QMAKE_CXXFLAGS += -Wa,-xgot
+QMAKE_CFLAGS += -Wa,-xgot
+
 win32:!shared:CONFIG += staticlib
 
 win32-borland {
